modules/core/test/test_io.cpp

```diff
@@ -1092,6 +1092,38 @@ TEST(Core_InputOutput, filestorage_yaml_advanvced_type_heading)
     ASSERT_EQ(cv::norm(inputMatrix, actualMatrix, NORM_INF), 0.);
 }
 
+TEST(Core_InputOutput, filestorage_matx_io)
+{
+    Matx33d matxTest(1.234, 2, 3, 4, 5, 6, 7, 8, 9.876);
+    Matx32d matxTestWrongSize(1, 2, 3, 4, 5, 6);
+    Mat normalMat = Mat::eye(3, 3, CV_64F);
+
+    std::vector<FileStorage::Mode> formats;
+    formats.push_back(FileStorage::Mode::FORMAT_JSON);
+    formats.push_back(FileStorage::Mode::FORMAT_XML);
+    formats.push_back(FileStorage::Mode::FORMAT_YAML);
+
+    for(size_t i = 0; i < formats.size(); i++)
+    {
```

Comment:
It is better to create multiple tests instead of single monster. See Google Test manual about parameterized tests.

Or at least add `SCOPED_TRACE(cv::format("format=%d", (int)formats[i]));` into for body (to make error messages more useful).

Reply:
I can split the three inner tests (matxTest, matxTestWrongSize and normalMat) into separate tests. I can also drop the test for the three formats into just one (i.e. JSON) if it's not adding value, though other tests seem to test for all three in a single run.